pkg/discovery/k8s_discovery_client.go

```diff
@@ -124,9 +170,40 @@ func (dc *K8sDiscoveryClient) Discover(accountValues []*proto.AccountValue) (*pr
 	entityDtos = append(entityDtos, podEntityDtos...)
 	entityDtos = append(entityDtos, appEntityDtos...)
 	entityDtos = append(entityDtos, serviceEntityDtos...)
-	discoveryResponse := &proto.DiscoveryResponse{
-		EntityDTO: entityDtos,
+
+	return entityDtos, nil
+}
+
+// A testing function for invoking discovery process with the new discovery framework.
+func (dc *K8sDiscoveryClient) DiscoverWithNewFramework() ([]*proto.EntityDTO, error) {
+	listOption := metav1.ListOptions{
+		LabelSelector: labels.Everything().String(),
+		FieldSelector: fields.Everything().String(),
+	}
+	nodeList, err := dc.config.kubeClient.Nodes().List(listOption)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list all nodes in the cluster: %s", err)
+	}
+	var nodes []*api.Node
+	for _, n := range nodeList.Items {
+		node := n
+		nodes = append(nodes, &node)
 	}
 
-	return discoveryResponse, nil
+	workerCount := dc.dispatcher.Dispatch(nodes, workerCount)
```

Comment:
not necessary to transfer the []api.Node to  []*apiNode.
Slices are reference of array, so you can pass the nodeList.Items into dc.dispatcher.Dispatch() directly without worrying about efficiency.
And in the function Dispatch(), you can use the slices, without the necessary to change the content to points.
 

I get the conclusion based on this:
http://openmymind.net/The-Minimum-You-Need-To-Know-About-Arrays-And-Slices-In-Go/

https://play.golang.org/p/dT24ThNddw